internal/service/streamprivatelinkendpoint/resource_test.go

```diff
@@ -1,34 +1,290 @@
+		display_name     = "terraform-test-private-link-network"
+		cloud            = %[2]q
+		region           = %[3]q
+		connection_types = ["PRIVATELINK"]
+		zones            = ["use1-az1", "use1-az2", "use1-az4"]
+		environment {
+			id = data.confluent_environment.default_environment.id
+		}
+		dns_config {
+			resolution = "PRIVATE"
+		}
+	}
+
+	resource "confluent_private_link_access" "aws" {
+		display_name = "terraform-test-aws-private-link-access"
+		aws {
+			account = %[4]q
+		}
+		environment {
+			id = data.confluent_environment.default_environment.id
+		}
+		network {
+			id = confluent_network.private-link.id
+		}
+	}
+
+	resource "confluent_kafka_cluster" "dedicated" {
+		display_name = "terraform-test-cluster"
+		availability = "MULTI_ZONE"
+		cloud        = confluent_network.private-link.cloud
+		region       = confluent_network.private-link.region
+		dedicated {
+			cku = 2
+		}
+		environment {
+			id = data.confluent_environment.default_environment.id
+		}
+		network {
+			id = confluent_network.private-link.id
+		}
+	}`, acc.ConfigConfluentProvider(), provider, region, awsAccountID)
+}
+
+func configDataConfluentDedicatedCluster(networkID, privatelinkAccessID string) string {
+	return fmt.Sprintf(`
+	%[1]s
+
+	data "confluent_environment" "default_environment" {
+  		display_name = "default"
+	}
+
+	data "confluent_network" "private-link" {
+		id     = %[2]q
+		environment {
+			id = data.confluent_environment.default_environment.id
+		}
+	}
+
+	data "confluent_private_link_access" "aws" {
+		id = %[3]q
+		environment {
+			id = data.confluent_environment.default_environment.id
+		}
+	}`, acc.ConfigConfluentProvider(), networkID, privatelinkAccessID)
+}
+
+func getCompleteConfluentConfig(usesExistingConfluentCluster bool, projectID, provider, region, vendor, awsAccountID, networkID, privatelinkAccessID string) string {
+	if usesExistingConfluentCluster {
+		configBasicUsingDatasources := strings.ReplaceAll(configBasic(projectID, provider, region, vendor, true), "confluent_network.private-link", "data.confluent_network.private-link")
+		configBasicUsingDatasourcesWithoutDependsOnCluster := strings.ReplaceAll(configBasicUsingDatasources, "confluent_kafka_cluster.dedicated", "")
+		return configDataConfluentDedicatedCluster(networkID, privatelinkAccessID) + configBasicUsingDatasourcesWithoutDependsOnCluster
+	}
+	return configNewConfluentDedicatedCluster(provider, region, awsAccountID) + configBasic(projectID, provider, region, vendor, true)
+}
+
+func checksStreamPrivatelinkEndpoint(projectID, provider, region, vendor string, dnsSubdomainsCheck bool) resource.TestCheckFunc {
+	checks := []resource.TestCheckFunc{checkExists()}
+	attrMap := map[string]string{
+		"project_id":    projectID,
+		"provider_name": provider,
+		"region":        region,
+		"vendor":        vendor,
+	}
+	pluralMap := map[string]string{
+		"project_id": projectID,
+		"results.#":  "1",
+	}
+	attrSet := []string{
```

Comment:
nit: i would put slice values in a line,  ok for maps to be one key/value pair per line